bundles/mapping/mapwfs2/plugin/WfsVectorLayerPlugin/impl/MvtLayerHandler.ol.js

```diff
@@ -92,14 +112,92 @@ export class MvtLayerHandler extends AbstractLayerHandler {
         // Add +1 to make a layer visible at the zoom level.
         return zoomLevelScale + 1;
     }
+
     /**
      * Adds event listeners to ol-layers
      * @param {string|number} layerId
      * @param {ol/source/VectorTile} oskariLayer
      */
-    _registerLayerEvents (layerId, source) {
-        source.on('tileloadstart', () => this.plugin.getMapModule().loadingState(layerId, true));
-        source.on('tileloadend', () => this.plugin.getMapModule().loadingState(layerId, false));
-        source.on('tileloaderror', () => this.plugin.getMapModule().loadingState(layerId, null, true));
+    _registerLayerEvents(layerId, source) {
+
+        const tileCounter = this.counters.get(layerId);
+
+        source.on('tileloadstart', () => {
+            tileCounter.started++;
+            this.plugin.getMapModule().loadingState(layerId, true);
+            this._sendWFSStatusChangedEvent(layerId, tileCounter, 'tileloadstart');
+        });
+        source.on('tileloadend', () => {
+            tileCounter.success++;
+            this.plugin.getMapModule().loadingState(layerId, false);
+            this._sendWFSStatusChangedEvent(layerId, tileCounter, 'tileloadend');
+        });
+        source.on('tileloaderror', () => {
+            tileCounter.error++;
+            this.plugin.getMapModule().loadingState(layerId, null, true);
+            this._sendWFSStatusChangedEvent(layerId, tileCounter, 'tileloaderror');
+        });
+    }
+
+    _sendWFSStatusChangedEvent(layerId, tileCounter, tileLoadStatusEvent) {
+
+        switch (tileLoadStatusEvent) {
+            case 'tileloadstart':
+
+                if (tileCounter.started === 1) {
+                    super.sendWFSStatusChangedEvent(layerId, 'loading');
+                    this._setTimer(layerId, tileCounter);
+                }
+                break;
+            case 'tileloadend':
+            case 'tileloaderror':
+
+                if (this._allStartedTileLoadingsFailed(tileCounter)) {
+                    this._resetTimer(layerId);
+                    super.sendWFSStatusChangedEvent(layerId, 'error');
+                    this._resetCounter(tileCounter);
+                } else if (this._allStartedTileLoadingsAreDone(tileCounter)) {
```

Comment:
Have you tried what happens if you remove the layer from the map when it's still loading? You probably need to reset the timer and counters on layer removal as well as a cleanup.

Reply:
I didn't test mentioned scenario but i will do that and include needed fixes to this PR once implemented.

Reply:
Would probably work as well if you always reset things in _initializeCountersForLayerIfNeeded() instead of just when missing.

Reply:
I made change to reset counters every time layer is added to map. Also added some other fixes to problems noticed during error case testing and during testing of layer removal / re-adding layer.

Reply:
According to testing, this proposed solution works. However more advanced solution could be tracking counters and timers per layer instance instead of per layerid since user can add and remove same layer multiple times while tileloadings maybe pending in background.